pulp_ansible/app/models.py

```diff
@@ -417,3 +417,44 @@ class AnsibleDistribution(Distribution):
 
     class Meta:
         default_related_name = "%(app_label)s_%(model_name)s"
+
+
+class RepositoryCollectionVersionIndex(models.Model):
+    """
+    An unmanaged model that uses a postgres view to index collectionversions.
+    """
+
+    # the id is a concat|aggregate of the repoid and the collectionversionid
+    id = models.CharField(max_length=255, primary_key=True)
+
+    # the view has the primary key for the repository and this auto-adds
+    # the relevant repository object to the model
+    repository = models.ForeignKey(AnsibleRepository, on_delete=models.DO_NOTHING)
+
+    # the view has the primary key for the collectionversion and this auto-adds
+    # the relevant collectionversion object to the model
+    collectionversion = models.ForeignKey(CollectionVersion, on_delete=models.DO_NOTHING)
+
+    # computed and stored in the view
+    reponame = models.CharField(max_length=255)
+
+    # computed and stored in the view
+    namespace = models.CharField(max_length=255)
+
+    # computed and stored in the view
+    name = models.CharField(max_length=255)
+
+    # computed and stored in the view
+    version = models.CharField(max_length=255)
+
+    # the number of deprecations found for this namespace.name
+    # if >0, then it is "deprecated"
+    deprecation_count = models.IntegerField()
+
+    # the count of signatures that are attached to the collectionversion
+    # which are also in the repository as content
+    sig_count = models.IntegerField()
+
+    class Meta:
```

Comment:
I guess if you add `abstract = True` Django will stop generating migrations and CI will pass

Reply:
I was reading that Django would generate a migration even for unmanaged models (not sure why), but I think it has something to do with foreign keys.

Reply:
Nevermind... I just tried and this does the job but looks like django won't populate the manager i.e. objects making it non queryable. 

ACtually you need to run makemigrations and commit the generated migration for the view, it is a **dummy** migration but is required.

Reply:
Got it, ty!